sdk/core/core/src/az_span_builder.c

```diff
@@ -23,6 +25,59 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
   return AZ_OK;
 }
 
+AZ_INLINE uint8_t _az_decimal_to_ascii(uint8_t const d) {
+  assert(d < 10);
```

Comment:
wondering...  why assert would be better than our az_contract here?

Reply:
it is a single line function that is only called a few lines below, assert shoud never be triggered because we should do mod 10. It is only if things go super wrong and whoever modifies it (why?) basically not paying attention at all.
AZ_CONTRACT seems a bit overkill, and to me it more looks like its purpose is to check the externally passed arguments whose values are beyond our control;